### ISSUE TITLE
feature: Support PEP 695 Type Parameter Syntax

### ISSUE THREAD

**Issue by viccie30** [mkdocstrings/griffe#342]
### Is your feature request related to a problem? Please describe.
Since Python 3.12 it is possible to create compile-time only type aliases with the new [type statement](https://docs.python.org/3.12/reference/simple_stmts.html#type). Sadly, griffe does not appear to parse these new statements yet. It also does not seem to parse [type parameters](https://docs.python.org/3.12/reference/compound_stmts.html#type-params) in class and function definitions.

### Describe the solution you'd like
I would like for griffe to parse these new constructs.

### Describe alternatives you've considered
N/A, I prefer to continue using mkdocstring instead of alternatives.

### Additional context
N/A

**Comment by pawamoy** [comment]
Yup, it's time we support the new language features :slightly_smiling_face:
